ros1_bridge/__init__.py

```diff
@@ -596,27 +596,37 @@ def determine_common_services(
                 break
             for i, ros1_field in enumerate(ros1_fields[direction]):
                 ros1_type = ros1_field[0]
-                ros2_type = str(ros2_fields[direction][i].type)
+                ros2_type = ros2_fields[direction][i].type
                 ros1_name = ros1_field[1]
                 ros2_name = ros2_fields[direction][i].name
-                if ros1_type != ros2_type or ros1_name != ros2_name:
+                if ros1_type != str(ros2_type) or ros1_name != ros2_name:
                     # if the message types have a custom mapping their names
                     # might not be equal, therefore check the message pairs
-                    if (ros1_type, ros2_type) not in message_string_pairs:
+                    if (ros1_type, str(ros2_type)) not in message_string_pairs:
                         match = False
                         break
+                if not ros2_type.array_size and not ros2_type.is_upper_bound:
+                    array_size = ros2_type.array_size if ros2_type.is_upper_bound else -1
```

Comment:
I don't understand what the purpose of these two lines is. After the condition in line 608 `ros2_type.is_upper_bound` must be `False` in order to enter the `if` block. So in this line that means `array_size` is always set to `-1` - never to `ros2_type.array_size` (which based on the condition in line 608 would need to be `None` anyway).